src/main/java/Duke.java

```diff
@@ -1,10 +1,71 @@
+
 public class Duke {
+    public static final int TODO_DESCRIPTION_START_INDEX = 5;
+    public static final int DEADLINE_DESCRIPTION_START_INDEX = 9;
+    public static final int DEADLINE_BY_SPACE_LENGTH = 4;
+    public static final int EVENT_DESCRIPTION_START_INDEX = 6;
+    public static final int EVENT_FROM_SPACE_LENGTH = 6;
+    public static final int EVENT_TO_SPACE_LENGTH = 4;
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        Scanner in = new Scanner(System.in);
+        System.out.println("Hello! I'm Gary");
+        System.out.println("What can I do for you?");
+
+        Task[] todos = new Task[100];
+        int todosCount = 0;
+
+        String line;
+        line = in.nextLine();
+        String[] lineWords;
+
+        while (!(line.toUpperCase().contains("BYE"))) {
+            lineWords = line.split(" ");
+            if (line.equalsIgnoreCase("LIST")) {
+                System.out.println("Here are the tasks in your list:");
+                for(int i = 0; i < todosCount; i += 1) {
+                    todos[i].printTask(i);
+                }
+            } else if (lineWords[0].equalsIgnoreCase("MARK")) {
+                todos[Integer.parseInt(lineWords[1]) - 1].markAsDone();
+                System.out.println("Nice! I've marked this task as done:");
+                System.out.println("  "
+                        + "[X] "
+                        + todos[Integer.parseInt(lineWords[1]) - 1].getTaskDescription());
+            } else if (lineWords[0].equalsIgnoreCase("UNMARK")) {
+                todos[Integer.parseInt(lineWords[1]) - 1].unmarkAsDone();
+                System.out.println("Ok, I've marked this task as not done yet:");
+                System.out.println("  "
+                        + "[ ] "
+                        + todos[Integer.parseInt(lineWords[1]) - 1].getTaskDescription());
+            } else {
+                if (lineWords[0].equalsIgnoreCase("TODO")) {
+                    todos[todosCount] = new Todo(line.substring(TODO_DESCRIPTION_START_INDEX));
+                } else if (lineWords[0].equalsIgnoreCase("Deadline")) {
+                    int bySlashIndex = line.indexOf("/");
+                    String deadlineDescription = line.substring(DEADLINE_DESCRIPTION_START_INDEX, bySlashIndex);
+                    String deadlineBy = line.substring(bySlashIndex + DEADLINE_BY_SPACE_LENGTH);
+                    todos[todosCount] = new Deadline(deadlineDescription, deadlineBy);
+                } else {
+                    //BUG IN FROM AND TO
+                    int fromSlashIndex = line.indexOf("/");
+                    String substringBeforeFrom = line.substring(0, fromSlashIndex + 1);
```

Comment:
There is quite some nasty if coditional nesting here, so I believe it's better to break it up to another method (e.g. `AddItemsToTheList(lineWords[0])`) that calls all these conditionals.

```
} else {
                if (lineWords[0].equalsIgnoreCase("TODO")) {
                    todos[todosCount] = new Todo(line.substring(TODO_DESCRIPTION_START_INDEX));
                } else if (lineWords[0].equalsIgnoreCase("Deadline")) {
                    int bySlashIndex = line.indexOf("/");
                    String deadlineDescription = line.substring(DEADLINE_DESCRIPTION_START_INDEX, bySlashIndex);
                    String deadlineBy = line.substring(bySlashIndex + DEADLINE_BY_SPACE_LENGTH);
                    todos[todosCount] = new Deadline(deadlineDescription, deadlineBy);
                } else {
                    //BUG IN FROM AND TO
                    int fromSlashIndex = line.indexOf("/");
                    String substringBeforeFrom = line.substring(0, fromSlashIndex + 1);

                    int toSlashIndex = substringBeforeFrom.length()
                            + line.substring(fromSlashIndex + 1).indexOf("/");
                    String eventDescription = line.substring(EVENT_DESCRIPTION_START_INDEX, fromSlashIndex);
                    String eventFrom = line.substring(fromSlashIndex + EVENT_FROM_SPACE_LENGTH, toSlashIndex);
                    String eventTo = line.substring(toSlashIndex + EVENT_TO_SPACE_LENGTH);
                    todos[todosCount] = new Event(eventDescription, eventFrom, eventTo);
                }
                todosCount += 1;
                todos[todosCount - 1].printAdd(todosCount);
            }
```